testApp/src/Components/Sidebar.svelte

```diff
@@ -0,0 +1,91 @@
+<!-- Sidebar.svelte -->
+<script lang="ts">
+    import { createEventDispatcher, onMount, onDestroy } from 'svelte';
+    export let sidebarOpen: boolean;
+    export let selectFranchise: (id: number) => void;
+
+    const dispatch = createEventDispatcher();
+
+    function handleSelectFranchise(id) {
+        selectFranchise(id);
+        sidebarOpen = !sidebarOpen;
+    }
+
+    function handleClickOutside(event) {
+        if (!event.target.closest('.sidebar, .fa-bars')) {
```

Comment:
What happens when someone decides to rename the `sidebar` class in the template, of move around some elements? This could probably use some tighter coupling, for example, using a variable which holds the sidebar class would be an improvement already.

Reply:
I implemented your last feedback to change it to pressing the escape button so i removed this part but i do understand what you mean!

https://github.com/BowenGeense/MovieShowcase/pull/3/files#diff-5fa8bc1e300230c2638d059cc9cd608ab21c79c6733942d5ddb7ca00bae85311